src/storage/index/hash_index_builder.cpp

```diff
@@ -126,12 +121,13 @@ template<typename T>
 template<bool IS_LOOKUP>
 bool HashIndexBuilder<T>::lookupOrExistsInSlotWithoutLock(
     Slot<T>* slot, const uint8_t* key, offset_t* result) {
-    for (auto entryPos = 0u; entryPos < slotCapacity; entryPos++) {
-        if (!slot->header.isEntryValid(entryPos)) {
-            continue;
-        }
+    auto guard = inMemOverflowFile ?
+                     std::make_optional<MutexGuard<InMemFile>>(inMemOverflowFile->lock()) :
+                     std::nullopt;
+    auto memFile = guard ? guard->get() : nullptr;
```

Comment:
Can you open an issue to optimize locks on memFile for string keys insertion? We can discuss how to do this. One way is to let each thread grab a page at a time, so reducing the total num of lock acquisitions.